apps/src/lib/kits/maker/PlaygroundComponents.js

```diff
@@ -54,6 +54,64 @@ export function initializeCircuitPlaygroundComponents(board) {
 }
 
 /**
+ * De-initializes any Johnny-Five components that might have been created
+ * by createCircuitPlaygroundComponents
+ * @param {Object} components - map of components, as originally returned by
+ *   createCircuitPlaygroundComponents.  This object will be mutated: Destroyed
+ *   components will be removed. Additional members of this object will be
+ *   ignored.
+ */
+export function destroyCircuitPlaygroundComponents(components) {
+  if (components.colorLeds) {
+    components.colorLeds.forEach(led => led.stop());
+  }
+  delete components.colorLeds;
+
+  if (components.led) {
+    components.led.stop();
+  }
+  delete components.led;
+
+  // No reset needed for five.Switch
+  delete components.toggleSwitch;
+
+  if (components.buzzer) {
+    components.buzzer.stop();
+  }
+  delete components.buzzer;
+
+  if (components.soundSensor) {
+    components.soundSensor.disable();
+  }
+  delete components.soundSensor;
+
+  // five.Thermometer makes an untracked setInterval call, so it can't be
+  // cleaned up.
+  // TODO: Fork / fix johnny-five Thermometer to be clean-uppable
+  // See https://github.com/rwaldron/johnny-five/issues/1297
+  delete components.tempSensor;
```

Comment:
This TODO is going to be somewhat complicated; I'll like to punt on it until a future PR.